src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -934,17 +934,31 @@ private IEnumerable<SecurityKey> GetContentEncryptionKeys(JsonWebToken jwtToken,
                 return keys;
 
             var unwrappedKeys = new List<SecurityKey>();
+            // keep track of exceptions thrown, keys that were tried
+            var exceptionStrings = new StringBuilder();
+            var keysAttempted = new StringBuilder();
             foreach (var key in keys)
             {
-                if (key.CryptoProviderFactory.IsSupportedAlgorithm(jwtToken.Alg, key))
+                try
                 {
-                    var kwp = key.CryptoProviderFactory.CreateKeyWrapProviderForUnwrap(key, jwtToken.Alg);
-                    var unwrappedKey = kwp.UnwrapKey(Base64UrlEncoder.DecodeBytes(jwtToken.EncryptedKey));
-                    unwrappedKeys.Add(new SymmetricSecurityKey(unwrappedKey));
+                    if (key.CryptoProviderFactory.IsSupportedAlgorithm(jwtToken.Alg, key))
+                    {
+                        var kwp = key.CryptoProviderFactory.CreateKeyWrapProviderForUnwrap(key, jwtToken.Alg);
+                        var unwrappedKey = kwp.UnwrapKey(Base64UrlEncoder.DecodeBytes(jwtToken.EncryptedKey));
+                        unwrappedKeys.Add(new SymmetricSecurityKey(unwrappedKey));
+                    }
                 }
+                catch (Exception ex)
+                {
+                    exceptionStrings.AppendLine(ex.ToString());
+                }
+                keysAttempted.AppendLine(key.ToString() + " , KeyId: " + key.KeyId);
```

Comment:
Not sure if we need the keyId here, as SecurityKey.ToString() prints it too:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/19092cb44fd556084c4cef9dfef08ec38e2ee130/src/Microsoft.IdentityModel.Tokens/SecurityKey.cs#L92-L95

Reply:
You're right, we don't need KeyId. I made the change.